src/dimension/dimension_trait.rs

```diff
@@ -291,12 +291,11 @@ pub trait Dimension:
         let order = strides._fastest_varying_stride_order();
```

Comment:
Reading this function, a few lines above - it seems there is still the case of an 1D array with stride -1 and it will still say it's not contig :(

Reply:
That's right. I checked the 1d array in dimension_trait.rs:289 and it now gets the correct result.

Reply:
I just found a good way. Refer to the practice in test_contiguous, use as_slice_memory_order().is_some()